resize.js

```diff
@@ -25,356 +25,371 @@ var consumeToken;
-        } else {
-            log('info', 'Uploading of ' + key + ' went very well');
-            var url = config.get('aws.bucket_url') + '/' + key;
-            insertImage.run([fileName, resolutionX, resolutionY, supportedFileType(fileType), url], function (err) {
-                if (err) {
-                    console.error(err);
-                }
-            });
+          console.error(err);
         }
-    });
+      });
+    }
+  });
 };
 image.upload = function (req, res) {
-    // Upload the RAW image to disk, stripped of its extension
-    // First check the token
-    var sentToken = req.headers['x-token'];
-    var matches = req.url.match(/^\/(.*)\.([^.]+)$/);
-    log('info', "Requested image upload for image_id " + matches[1] + " with token " + sentToken);
-    if (supportedFileType(matches[2])) {
-        // We support the file type
-        consumeToken.run([sentToken, matches[1]], function (err) {
-            if (!err && this.changes === 1) {
-                // And we support the filetype
-                log('info', 'Starting to write original file ' + matches[1]);
-                var form = new formidable.IncomingForm();
-                form.parse(req, function(err, fields, files) {
-                    res.writeHead(200, {'Content-Type': 'application/json'});
-                    res.write(JSON.stringify({'status': 'OK', 'id': matches[1]}));
-                    res.end();  
-                });
+  // Upload the RAW image to disk, stripped of its extension
+  // First check the token
+  var sentToken = req.headers['x-token'];
+  var matches = req.url.match(/^\/(.*)\.([^.]+)$/);
+  log('info', "Requested image upload for image_id " + matches[1] + " with token " + sentToken);
+  if (supportedFileType(matches[2])) {
+    // We support the file type
+    consumeToken.run([sentToken, matches[1]], function (err) {
+      if (!err && this.changes === 1) {
+        // And we support the filetype
+        log('info', 'Starting to write original file ' + matches[1]);
+        var form = new formidable.IncomingForm();
 
-                form.on('end', function(fields, files) {
-                    var temp_path = this.openedFiles[0].path;
+        form.parse(req, function (err, fields, files) {
+          var temp_path = files.image.path;
+          var destination_path = config.get('originals_dir') + '/' + matches[1];
+          console.log(err, files, destination_path);
 
-                    fs.copy(temp_path, config.get('originals_dir') + '/' + matches[1], function(err) {  
-                        if (err) {
-                            console.error(err);
-                        } else {
-                            log('info', 'Finished writing original file ' + matches[1]);
-                        }
-                    });
-                });
+          fs.copy(temp_path, destination_path, function (err) {
+            if (err) {
+              console.error(err);
```

Comment:
You just swallow the error here, the user won't get notified that anything gets wrong.